scripts/main/photo.js

```diff
@@ -1183,85 +1172,85 @@ photo.showDirectLinks = function (photoID) {
-		);
-	}
-	if (photo.json.size_variants.medium !== null) {
-		msg += buildLine(
-			`${lychee.locale["PHOTO_MEDIUM"]} (${photo.json.size_variants.medium.width}x${photo.json.size_variants.medium.height})`,
-			lychee.getBaseUrl() + photo.json.size_variants.medium.url
-		);
-	}
-	if (photo.json.size_variants.small2x !== null) {
-		msg += buildLine(
-			`${lychee.locale["PHOTO_SMALL_HIDPI"]} (${photo.json.size_variants.small2x.width}x${photo.json.size_variants.small2x.height})`,
-			lychee.getBaseUrl() + photo.json.size_variants.small2x.url
-		);
-	}
-	if (photo.json.size_variants.small !== null) {
-		msg += buildLine(
-			`${lychee.locale["PHOTO_SMALL"]} (${photo.json.size_variants.small.width}x${photo.json.size_variants.small.height})`,
-			lychee.getBaseUrl() + photo.json.size_variants.small.url
-		);
-	}
-	if (photo.json.size_variants.thumb2x !== null) {
-		msg += buildLine(
-			`${lychee.locale["PHOTO_THUMB_HIDPI"]} (${photo.json.size_variants.thumb2x.width}x${photo.json.size_variants.thumb2x.height})`,
-			lychee.getBaseUrl() + photo.json.size_variants.thumb2x.url
-		);
-	}
-	if (photo.json.size_variants.thumb !== null) {
-		msg += buildLine(
-			`${lychee.locale["PHOTO_THUMB"]} (${photo.json.size_variants.thumb.width}x${photo.json.size_variants.thumb.height})`,
-			lychee.getBaseUrl() + photo.json.size_variants.thumb.url
-		);
-	}
-	if (photo.json.live_photo_url) {
-		msg += buildLine(` ${lychee.locale["PHOTO_LIVE_VIDEO"]} `, lychee.getBaseUrl() + photo.json.live_photo_url);
-	}
+	const localizations = {
+		original: lychee.locale["PHOTO_FULL"],
+		medium2x: lychee.locale["PHOTO_MEDIUM_HIDPI"],
+		medium: lychee.locale["PHOTO_MEDIUM"],
+		small2x: lychee.locale["PHOTO_SMALL_HIDPI"],
+		small: lychee.locale["PHOTO_SMALL"],
+		thumb2x: lychee.locale["PHOTO_THUMB_HIDPI"],
+		thumb: lychee.locale["PHOTO_THUMB"],
+	};
+
+	const showDirectLinksDialogBody =
+		'<form class="photo-links">' +
+		buildLine("view") +
+		'</form><p></p><form class="photo-links">' +
+		Object.entries(localizations).reduce((html, [type]) => html + buildLine(type), "") +
+		buildLine("live") +
+		"</form>";
+
+	/**
+	 * @param {ModalDialogFormElements} formElements
+	 * @param {HTMLDivElement} dialog
+	 */
+	const initShowDirectLinksDialog = function (formElements, dialog) {
+		formElements.view.value = photo.getViewLink(photoID);
+		formElements.view.previousElementSibling.textContent = lychee.locale["PHOTO_VIEW"];
+		formElements.view.nextElementSibling.title = "Copy to clipboard";
+		dialog.querySelector("p").textContent = lychee.locale["PHOTO_DIRECT_LINKS_TO_IMAGES"];
+
+		for (const type in localizations) {
+			/** @type {?SizeVariant} */
+			const sv = photo.json.size_variants[type];
+			if (sv !== null) {
+				formElements[type].value = lychee.getBaseUrl() + sv.url;
+				formElements[type].previousElementSibling.textContent = localizations[type] + " (" + sv.width + "×" + sv.height + ")";
+				formElements[type].nextElementSibling.title = "Copy to clipboard";
```

Comment:
Same here (localization).

Reply:
Fixed in https://github.com/LycheeOrg/Lychee-front/commit/7065ddba2d9db477f9940c9ce6a875fd4e477840.